src/NetscriptFunctions/Bladeburner.ts

```diff
@@ -325,9 +325,8 @@ export function NetscriptBladeburner(): InternalAPI<INetscriptBladeburner> {
       const bladeburner = getBladeburner(ctx);
       return Math.round(bladeburner.storedCycles / 5) * 1000;
     },
-    nextCycle: (ctx) => async () => {
-      const bladeburner = getBladeburner(ctx);
-      return new Promise<number>((res) => bladeburner.resolvers.push(res));
+    nextCycle: () => async () => {
```

Comment:
Optional: A minor thing, these don't need to be async, they're just regular functions that return a promise. I think having them be `async` is technically observable in terms of exactly how the resulting promise is wrapped, but not in a way anyone should be depending on.